src/helpers/query-all-survey-answers.ts

```diff
@@ -0,0 +1,25 @@
+import MyDataHelps, { Guid, SurveyAnswer, SurveyAnswersPage, SurveyAnswersQuery } from "@careevolution/mydatahelps-js";
+
+export default async function (props: SurveyAnswersQuery): Promise<SurveyAnswer[]> {
+
+    async function getSurveyAnswers(): Promise<SurveyAnswer[]> {
+        let dataPage = await getSurveyDataPage();
+        let allData = dataPage.surveyAnswers;
+        while (dataPage.nextPageID) {
+            dataPage = await getSurveyDataPage(dataPage.nextPageID);
+            allData = allData.concat(dataPage.surveyAnswers);
+        }
+        return allData;
+    }
+
+    async function getSurveyDataPage(pageID?: Guid): Promise<SurveyAnswersPage> {
+        var queryParameters: SurveyAnswersQuery = props;
```

Comment:
Do we need the extra variable here?  I think you can just set the `pageID` on `props` in the conditional below.

If the intent was to not modify the `props` object, I think this may need to be `{...props}` instead.